### ISSUE TITLE
sample_profile is missing sequenced samples without mutations

### ISSUE THREAD

**Issue by jjgao** [cBioPortal/cbioportal#4109]
Samples are added to sample_profile when importing the MAF, but only samples with mutated are added.

https://github.com/cBioPortal/cbioportal/blob/master/core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java#L431

- [x] Look for samples in `#sequenced_samples:` line. If available, add these first.
- [x] Is it possible to write a migrations script to fill the missing samples based on `_sequenced` sample list?

**Comment by jjgao** [comment]
`sample_profile` is now used in the API.

Fixing this is also a step towards refactoring sample list (connecting sample list to profile).

**Comment by jjgao** [comment]
cc'ing @ersinciftci and @adamabeshouse. Once that's done, let's 

- [ ] have an API to expose the `sample_profile` data
- [ ] use that API for determine whether a sample is profiled (related to https://github.com/cBioPortal/cbioportal/issues/3965)
- [ ] remove `wholeExomeSequenced` from gene_panel end point

**Comment by n1zea144** [comment]
@jjgao @ritikakundra I think I know the answer to this one , but wanted your feedback - there is code in this file that creates samples/patients if they are in the MAF but not in the clinical files.  I could take this out and throw an exception instead.  Do you think its a good time for this (probably not)?

Whatever we decide, I will have to add similar behavior for sample ids that are referenced in the sequenced_samples list in the header.

**Comment by jjgao** [comment]
@n1zea144 I think that's a good idea. Do you think we should throw an exception and stop the import process or continue?

Not sure if others always have all samples in the clinical data file? cc'ing @pieterlukasse 

Maybe there should be some rule implemented in validator too?

**Comment by jjgao** [comment]
@n1zea144 maybe we could have two pull requests? One for handling `#sequenced_samples` to address this issue.

And the other to take out the code you mentioned. I think this is low priority (since the current code doesn't hurt) and may require some testing.

**Comment by ritikakundra** [comment]
@jj validator does fail if it finds a sample in any genomic file but not in clinical sample file.

**Comment by n1zea144** [comment]
@jjgao I have a script to backfill samples missing from sample-profile table.  It makes some assumptions and is errors on the side of being conservative.  It checks for a single "cancerstudy_sequenced" case list and "cancerstudy_mutations" mutation profile.  If both are found, the script attempts to determine a valid gene panel or defaults to WES.  Once testing is complete, I will schedule a run of this on production.

**Comment by jjgao** [comment]
Thanks, @n1zea144! Please feel free to close this once the code is in production and the databases are migrated.

**Comment by jjgao** [comment]
@n1zea144 I think we can close this now?
